### ISSUE TITLE
Change either documentation or code of t.i.test.reactormixins.ReactorBuilder.requiredInterface

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#4527]
|[<img alt="cyli's avatar" src="https://avatars.githubusercontent.com/u/223200?s=50" width="50" height="50">](https://github.com/cyli)| @cyli reported|
|-|-|
|Trac ID|trac#4527|
|Type|defect|
|Created|2010-06-29 19:23:48Z|
|Branch|https://github.com/twisted/twisted/tree/reactorbuilder-requiredInterfaces-4527-2|

Currently, the documentation of ReactorBuilder says:

```
@cvar requiredInterfaces: A C{list} of interfaces which the reactor must
provide or these tests will be skipped.  The default, C{None}, means
that no interfaces are required.
```

However, the attribute is actually 'requiredInterface' and takes a single interface, rather than a list.  Either the docstring should change to reflect the actual attribute or the attribute (and the method buildReactor) should change to reflect the existing docstring.

<details><summary>Searchable metadata</summary>

```
trac-id__4527 4527
type__defect defect
reporter__cyli cyli
priority__normal normal
milestone__ 
branch__branches_reactorbuilder_requiredInterfaces_4527_2 branches/reactorbuilder-requiredInterfaces-4527-2
branch_author__cyli cyli
status__closed closed
resolution__fixed fixed
component__core core
keywords__easy easy
time__1277839428000000 1277839428000000
changetime__1277918828000000 1277918828000000
version__None None
owner__cyli cyli

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="cyli's avatar" src="https://avatars.githubusercontent.com/u/223200?s=50" width="50" height="50">](https://github.com/cyli)<a name="note_1"></a>|@cyli commented|
|-|-|

Note that t.i.test.test_fdset currently uses requiredInterface = IReactorFDSet, but t.i.test.test_process uses requiredInterfaces=[IReactorProcess], which from reading the ReactorBuilder code, seems like it wouldn't do anything.  So that's probably also a bug.

**Comment by twisted-trac** [comment]
|[<img alt="cyli's avatar" src="https://avatars.githubusercontent.com/u/223200?s=50" width="50" height="50">](https://github.com/cyli)<a name="note_2"></a>|@cyli commented|
|-|-|

Both could be provided for compatibility for now, but if so, one of them should probably be deprecated.

But since there are only two tests in trunk right now that use requiredInterface, it might not be too late to just jump one way or the other as quickly as possible (offer EITHER a single requiredInterface, or a list of requiredInterfaces).

**Comment by twisted-trac** [comment]
|[<img alt="cyli's avatar" src="https://avatars.githubusercontent.com/u/223200?s=50" width="50" height="50">](https://github.com/cyli)<a name="note_3"></a>|@cyli commented|
|-|-|

(In [29475]) Branching to 'reactorbuilder-requiredInterfaces-4527'

**Comment by twisted-trac** [comment]
|[<img alt="cyli's avatar" src="https://avatars.githubusercontent.com/u/223200?s=50" width="50" height="50">](https://github.com/cyli)<a name="note_4"></a>|@cyli commented|
|-|-|

Will merge forward to take into account two other branches that have been merged into trunk.

**Comment by twisted-trac** [comment]
|[<img alt="cyli's avatar" src="https://avatars.githubusercontent.com/u/223200?s=50" width="50" height="50">](https://github.com/cyli)<a name="note_5"></a>|@cyli commented|
|-|-|

(In [29482]) Changed ReactorBuilder's requiredInterface attribute to requiredInterfaces (which takes a list, as stated in the documentation)

refs [#4527](https://github.com/twisted/twisted/issues/4527)

**Comment by twisted-trac** [comment]
|[<img alt="cyli's avatar" src="https://avatars.githubusercontent.com/u/223200?s=50" width="50" height="50">](https://github.com/cyli)<a name="note_6"></a>|@cyli commented|
|-|-|

(In [29483]) Branching to 'reactorbuilder-requiredInterfaces-4527-2'

**Comment by twisted-trac** [comment]
|[<img alt="cyli's avatar" src="https://avatars.githubusercontent.com/u/223200?s=50" width="50" height="50">](https://github.com/cyli)<a name="note_7"></a>|@cyli commented|
|-|-|

(In [29485]) Changed usage of requiredInterface in trunk to requiredInterfaces

refs [#4527](https://github.com/twisted/twisted/issues/4527)

**Comment by twisted-trac** [comment]
|[<img alt="cyli's avatar" src="https://avatars.githubusercontent.com/u/223200?s=50" width="50" height="50">](https://github.com/cyli)<a name="note_8"></a>|@cyli removed owner|
|-|-|

http://buildbot.twistedmatrix.com/boxes-supported?branch=/branches/reactorbuilder-requiredInterfaces-4527-2

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_9"></a>|@exarkun set owner to @cyli|
|-|-|

Looks great.  Please merge!

**Comment by twisted-trac** [comment]
|[<img alt="cyli's avatar" src="https://avatars.githubusercontent.com/u/223200?s=50" width="50" height="50">](https://github.com/cyli)<a name="note_10"></a>|@cyli set status to `closed`|
|-|-|

(In [29520]) Merge reactorbuilder-requiredInterfaces-4527-2 : correct ReactorBuilder to be consistent with documentation

Author: cyli
Reviewer: exarkun
Fixes: [#4527](https://github.com/twisted/twisted/issues/4527)

ReactorBuilder had attribute requiredInterface, which is a single interface, while documentation said that it had attribute requiredInterfaces, which is a list of interfaces.  Changed ReactorBuilder to be consistent with its documentation.
